targets/TARGET_Cypress/TARGET_PSOC6/analogin_api.c

```diff
@@ -126,18 +131,22 @@ void analogin_init(analogin_t *obj, PinName pin)
 
     sar = pinmap_peripheral(pin, PinMap_ADC);
     if (sar != (uint32_t)NC) {
-        if (cy_reserve_io_pin(pin)) {
+	
+	if ((0 != cy_reserve_io_pin(pin)) && !sar_initialized) {
             error("ANALOG IN pin reservation conflict.");
         }
-        obj->base = (SAR_Type*)CY_PERIPHERAL_BASE(sar);
+		
```

Comment:
Nitpick - trailing whitespace here and in some other places in this commit

Reply:
@c1728p9: applied astyle to the HAL implementation.